src/services/FirebaseService.ts

```diff
@@ -114,16 +130,34 @@ export class FirebaseService {
    * Requests custom token for the address from Firebase function and signs in to Firebase
    */
   private async signInToFirebase(address: string): Promise<UserCredential> {
+    if (firebaseAuth.currentUser && firebaseAuth.currentUser.uid === address) {
+      return;
+    }
+
     // Signs out from Firebase in case another user was authenticated
     // (could happen when user disconnect and connect a new wallet)
     await signOut(firebaseAuth);
 
-    // Requests a custom Firebase Token used for signing in to Firebase, from our Firebase Function
-    const token = await this.createCustomToken(address);
+    const messageToSign = await this.getMessageToSign(address);
+
+    let signature: string;
+    try {
+      this.eventAggregator.publish("firebase.signature", true);
+      signature = await this.ethereumService.getDefaultSigner().signMessage(messageToSign);
+      this.eventAggregator.publish("firebase.signature", false);
```

Comment:
Nit: Can this be moved to a `finally` block? (also the one public in the `catch` block could be removed then?)